datageneration/tests/test_relation_generator.py

```diff
@@ -29,7 +38,58 @@ def test_within_radius(self):
+        except AssertionError:
+            raise RuntimeError('The test should not be failed! Something is wrong.')
+
+        # case 2, one area entity and the others are ...
+        area_entity = [Entity(id=0, is_area=True, name='astro station', type='nwr', properties=[])]
+        point_entities_connecting_to_area_entity = [[Entity(id=1, is_area=False, name='block', type='nwr',
+                                                           properties=[
+                                                               Property(name='height', operator='=',
+                                                                        value='0.6 m')]),
+                                                    Entity(id=2, is_area=False, name='scuba center', type='nwr',
+                                                           properties=[])]]
+
+        relations = self.relation_generator.generate_relation_with_contain_helper(drawn_area_entities=area_entity,
+                    point_entities_connecting_to_area_entity=point_entities_connecting_to_area_entity, add_dist=False)
+        expected_relations = [Relation(type='contains', source=0, target=1, value=None),
+                              Relation(type='contains', source=0, target=2, value=None)]
+
+        self.assertEqual(relations, expected_relations)
+
+        # Check if distance value was added
+        relations = self.relation_generator.generate_relation_with_contain_helper(drawn_area_entities=area_entity,
+                    point_entities_connecting_to_area_entity=point_entities_connecting_to_area_entity, add_dist=True)
+        self.assertTrue(relations[0].value != None)
+
+
+        # Check the combination with "individual distances"
+        area_entity = [Entity(id=0, is_area=True, name='astro station', type='nwr', properties=[])]
+        point_entities_connecting_to_area_entity = [[
+            Entity(id=1, is_area=False, name='scuba center', type='nwr', properties=[])]]
+
+        relations = self.relation_generator.generate_relation_with_contain_helper(
+                                drawn_area_entities=area_entity,
+                                point_entities_connecting_to_area_entity=point_entities_connecting_to_area_entity)
+
+        other_point_entities = [Entity(id=2, is_area=False, name='block', type='nwr',
+                                       properties=[Property(name='height', operator='=', value='0.6 m')])]
+        other_point_entities = point_entities_connecting_to_area_entity[0] + other_point_entities
+        other_entity_ids = [e.id for e in other_point_entities]
+
+        relations.extend(self.relation_generator.generate_individual_distances(entity_ids=other_entity_ids))
+        expected_relations = [Relation(type='contains', source=0, target=1, value=None),
+                              Relation(type='dist', source=1, target=2, value="100 m")]
+        relations[1].value = "100 m"
```

Comment:
You can use patch instead of assigning a value (direct asigning might be problematic in some cases) as it was done at the previous commit:
https://github.com/dw-innovation/kid2-spot-datageneration/blob/d41c955b055a351061e8ccbbb6646467ae43c31d/datageneration/tests/test_relation_generator.py#L68C1-L70C59

Reply:
This does not work any more I think, since "get_random_decimal_with_metric" is no longer part of class "RelationGenerator". It was in the "property_generator.py" file, even though I just moved it to "util.py" in the last commit.

Is there a way to run this patch if the method is not part of the class? Using the same code as in the previous commit does not work.